dev/com.ibm.ws.st.core/src/com/ibm/ws/st/core/internal/config/validation/ConfigurationValidator.java

```diff
@@ -58,6 +61,70 @@ public ValidationResult validate(IResource resource, int kind, ValidationState s
             Trace.logError("Error removing markers", e);
         }
 
+        // For defined runtimes, do not validate any server config files in a templates folder inside the runtime folder
+        IPath fileLocation = file.getLocation(); // Full system path of the file
+        IPath fileFullPath = file.getFullPath(); // Used to ensure that templates is a subfolder of the runtime
+        WebSphereRuntime[] webSphereRuntimes = WebSphereUtil.getWebSphereRuntimes();
+        int length = webSphereRuntimes.length;
+        for (int i = 0; i < length; i++) {
+            WebSphereRuntime runtime = webSphereRuntimes[i];
+            IPath runtimeLocation = runtime.getRuntimeLocation();
+            if (runtimeLocation.isPrefixOf(fileLocation) && fileFullPath.toString().contains("templates")) { //$NON-NLS-1$
+                return EMPTY_RESULT;
+            }
+        }
+
+        // Workaround to not validate server.xml files in templates folder in build plugins
+        // eg. target/liberty/wlp/templates
+        // eg. build/wlp/templates/
+        // Best guess to find a templates folder in a 'ghost' runtime
+        // This is the simplest check: if the file is in a path that contains wlp/templates
+        if (fileFullPath.toString().contains("wlp/templates")) { //$NON-NLS-1$
```

Comment:
The runtime folder does not have to be "wlp".

Reply:
Yes, this is the first quick check.  If it is, then we simply return.

Then we must rely on the second check, which is below the first check.

Reply:
Do you want to get rid of this first check and do only the second check?

Reply:
I think you should just get rid of the first check.

Reply:
Ok, done. Removed the first check.